compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -673,7 +673,9 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(using Context) {
           def apply(t: Type) = t match {
             case t: ThisType => thisProxy.getOrElse(t.cls, t)
             case t: TypeRef => paramProxy.getOrElse(t, mapOver(t))
-            case t: SingletonType => paramProxy.getOrElse(t, mapOver(t))
+            case t: SingletonType =>
+              if t.termSymbol.isAllOf(Inline | Param) then mapOver(t.widenTermRefExpr)
```

Comment:
What about use a case specifically for `TermRef`:

```scala
case t @ TermRef(NoPrefix, _) if t.termSymbol.isAllOf(Inline | Param) =>
    mapOver(t.widenTermRefExpr)
case t: SingletonType =>
    paramProxy.getOrElse(t, mapOver(t))
```

Reply:
That was my original code but den I redactores it to avoid the unnecessary extra type test.